shared/menubar/files-container.desktop.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import * as ConfigGen from '../actions/config-gen'
+import * as Tabs from '../constants/tabs'
+import * as FsTypes from '../constants/types/fs'
+import * as FsGen from '../actions/fs-gen'
+import {switchTo} from '../actions/route-tree'
+import {FilesPreview} from './files.desktop'
+import {remoteConnect, compose, type Dispatch} from '../util/container'
+
+const mapStateToProps = (state) => ({
+  _tlfRows: [
+    {path: FsTypes.stringToPath('/keybase/team/zila.test/abc')},
+    {path: FsTypes.stringToPath('/keybase/team/zila.test/def')},
+  ],
+})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  _onSelectPath: (path: FsTypes.Path) => {
+    dispatch(ConfigGen.createShowMain())
```

Comment:
It seems `ConfigGen.showMain` is created multiple times. Have you considered dispatching a single action here (and below for onViewAll), and in saga of the main window, do something like `Saga.actionToAction([FsGen.selectPath, FsGen.viewAll], ()=>Saga.put(ConfigGen.createShowMain()))`

Reply:
It's not really though...the creation only happens if the action is dispatched.

Reply:
Sorry, by "created multiple times", I meant this was declared multiple times (here are in `onViewAll` below). The alternative also has the benefit that these on* dispatchers only need to dispatch one action.

Reply:
I prefer it this way...why send the code reader to another place if they can see here exactly what is done? The suggestion will end up dispatching 4 actions instead of 3.